### ISSUE TITLE
Wrong error message for missing properties in types composed of the intersection of certain multiple unions 

### ISSUE THREAD

**Issue by ezheidtmann** [microsoft/TypeScript#53200]
# Bug Report

### 🔎 Search Terms

intersection of union required property error message

### 🕗 Version & Regression Information

This is the behavior in every version I tried, and I reviewed the FAQ for entries about it

I use the playground and observed the same behavior for versions 4.2 through 4.9

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?ts=4.9.5#code/C4TwDgpgBAggTgcygXilAFAKDQHygb2zWLDggDMAZAQwCMIAbAfgC4oBnYOASwDsEA3EWIcAFgHsA7gAUy5NuWoN2EISIC+RPIRFpSFGvQZtOPfmt3sJMuayhcArquFRNASiiYAZBhfao4sCiEHAKSioCrp4i-oHBofZwTpGaIm5CmJgAxuK8nFDkvCgYQXwIbPAIHsgAfATqGYXo+GJSshRhytDq6ZlNLVZtcp0qADQBQSFsjt1uQA)

### 💻 Code

```ts
type Arg =  (
  | {
      prefLabel?: string;
      showPref: false;
    }
  | {
      prefLabel: string;
      showPref?: true;
    }
) 
& (
    | { other: false; } 
    | { other: true; }
    );


const fn = (thing: Arg) => {};

fn({ showPref: false }); // <<< error on this line

fn({ showPref: false, other: true }); // <<< this is ok
```

### 🙁 Actual behavior

I see an error for `fn({ showPref: false });`:

```
Argument of type '{ showPref: false; }' is not assignable to parameter of type 'Arg'.
  Type '{ showPref: false; }' is not assignable to type '{ prefLabel: string; showPref?: true | undefined; } & { other: true; }'.
    Property 'prefLabel' is missing in type '{ showPref: false; }' but required in type '{ prefLabel: string; showPref?: true | undefined; }'.(2345)
input.tsx(7, 7): 'prefLabel' is declared here.
```

This is wrong because actually `other` is missing. The following line does not produce an error

### 🙂 Expected behavior

I would expect the error message to refer to the missing required property, not to an irrelevant union member

**Comment by RyanCavanaugh** [comment]
In the absence of finding a discriminant (which, long story short, this union doesn't have one) we end up reporting on a somewhat arbitrarily-chosen member of the target union. The error message isn't wrong, it's just not talking about the target type you wished it was.

**Comment by ezheidtmann** [comment]
In the line showing the error, `showPref` is not required but the error message says it is. That seems pretty wrong to me.

And if my understanding of unions is correct, `showPref` is not in the same union as `other`.

[Here's a variation](https://www.typescriptlang.org/play?ts=4.9.5#code/C4TwDgpgBAggTgcygXilAFAKDQHygb2zWLDggDMAZAQwCMIAbAfgC4oA7AVwFt64BuIsSgBnABYB7AO4AFMuTblqDEREHCAvkTyFhaUhRr0GbEcDgBLdgnV7x0uRVZRznNUKhaAlFEwAyDA8dKAlgMQg4RWVVfk9fYWDQ8MiXODdYrWEvQUxMAGMJdjMocnYUDDCrBDZ4BB9kAD4CDRzS9HxRSVl5KJVoDWzcto77bope1QAaKAMqOkY2AGZPQcxhzocekuiIadmjBahF6aSIif7VtfZ2jbGFVLc9+QOTKAAiOjy3k7Czh4vWtcRl1HPclH0fsk2K4LkA) that shows a different and still unhelpful message -- the error is about incompatible types for `prefLabel` but the type is fine if I add `other` to the object.

Regardless, if you believe the current behavior is not wrong, feel free to rephrase my issue to read "unhelpful and misleading error message" instead of "wrong."

**Comment by ezheidtmann** [comment]
In case it's helpful, [this is a variation with (what I hope is) a discriminant in the first union](https://www.typescriptlang.org/play?ts=4.9.5#code/C4TwDgpgBAggTgcygXilAFAKDQHygb2zWNEgC4oAiAQ0oG4jiow4IAzAGWoCMIAbAPwUAzsDgBLAHYIGTNMIAWAewDuABVZsKban2ERZxAL5E8hOVFIQKlbvUbEW7Lrz4ixUmQ-nL1moZZwAK4GDiYAlFCYAGQYDmZQSsAKEHDauvp0UEZRTAlJKWmBIVkmTOEMmJgAxkqSolBskigYyZ4U8AiRyAB8BEaVTej4luDWVLQANFCKqhrs6XrQRhVVQyNWNlMzvvNajRkQ0wWpFGIh2eFAA):

```
type Arg =  (
  | {
      type: "a";
      prefLabel?: string;
      showPref: false;
    }
  | {
      type: "b";
      prefLabel: string;
      showPref?: true;
    }
) 
& (
    | { other: false; } 
    | { other: true; }
    );


const fn = (thing: Arg) => {};

fn({ type: "a", showPref: false });

fn({ type: "a", showPref: false, other: true });
```

The error is still mentioning `prefLabel` in this case.
